### DISCUSSION TITLE
Remove cryptocurrencies category

### DISCUSSION THREAD

**Discussion by clarfonthey** [rust-lang/crates.io#6762] [Feature Requests]
Several years ago, it was very easy to give hopeful optimism to cryptocurrency as a new technology. That is no longer the case. There is overwhelming evidence that cryptocurrency has a direct impact on the planet's health, that it features a number of problems which make it unsuitable for general adoption, and that its only practical use is through financial schemes which aim to make a quick buck despite the other issues mentioned.

Given how Rust positions itself specifically as a language aimed at empowering people, particularly those in less fortunate positions than the average tech developer with a cushy salary, it seems extremely insensitive that the Rust project lends legitimacy to cryptocurrencies by including a dedicated category for them on crates.io.

This category was added with [minimal](https://github.com/rust-lang/crates.io/issues/1475) [discussion](https://github.com/rust-lang/crates.io/pull/1674) *and* can be superseded by the finance category which was [added after](https://github.com/rust-lang/crates.io/pull/5707) the cryptocurrencies category, or the cryptography category which predates it.

Some developers in the ecosystem have [already taken a stand](https://gitlab.com/crates.rs/crates.rs/-/issues/81) on this issue and explicitly delisted the cryptocurrencies category, despite [other developers' insistence](https://gitlab.com/crates.rs/crates.rs/-/issues/121) that such a category is legitimate and that taking a stand is outside the scope of Rust as a project.

I strongly disagree with this approach, since Rust already adopts a code of conduct which explicitly delineates inappropriate behaviour. Criticism of harassment of fellow developers, for example, is explicitly within the scope of the project. If a crate were uploaded to crates.io for the express purpose of harassing someone, it would likely be removed. I feel that taking a stand on a harmful technology like cryptocurrency is well within the scope of the project, but that's not even what I'm asking: I'm asking to simply stop giving cryptocurrency a special position on crates.io by removing its dedicated category. Rust can continue to be as neutral toward cryptocurrencies as they are to puzzle games, which also don't have a dedicated category, but still fit within other, less specific categories.

Cryptocurrency crates can continue to exist, but will not have their own dedicated category. They can continue to use keywords to signal that they're related to cryptocurrency, since these are not curated by the registry and work differently.

**Comment by castarco** [comment]
If there is a take that I would emphasise from your proposal, is this one:

> Rust can continue to be as neutral toward cryptocurrencies as they are to puzzle games, which also don't have a dedicated category, but still fit within other, less specific categories.

I totally agree. Web3, blockchain & cryptocurrencies don't deserve their own category.

**Comment by Turbo87** [comment]
We've talked about your request at the crates.io team meeting last week:

- crates.io categories were designed to be additive, which means that downstream system are assuming we will only add, but not remove, categories. removing a category would also cause issues for crates currently using a removed category, and we feel uneasy about creating such a precedent.
- if we remove the cryptocurrency category then these crates will most likely recategorize themselves as finance or cryptography crates, which makes it harder to find non-cryptocurrency crates in these categories. thus we feel like it might be better to keep the category so that it is easier to filter out this category of crates.
- speaking for me personally: I would prefer to completely get rid of categories, since they serve the same purpose as keywords, but with less flexibility and more support burden. I think a list of common/suggested/blessed keywords to standardize on would be much more valuable.

We hope you can understand our view and why we are choosing to not remove any specific category at this point.

**Reply by clarfonthey** [reply]
I appreciate the points here! While I don't 100% agree with the first two points, I think exploring the third would genuinely be a good idea. The way I see it, we could deprecate categories and then bump the limits of 5 keywords + 5 categories to 10 keywords. If a crate has categories, it would simply output a warning and then treat them as keywords on upload. For scoped categories, only the base name (and not the full path) would be included.

I might explore writing an RFC for this and will share it in other places first, but figured I'd mention here since I like the idea and would prefer to avoid explicitly curated categories.

**Comment by kornelski** [comment]
Oh my, this issue seems to be all about things I've done  :)

> if we remove the cryptocurrency category then these crates will most likely recategorize themselves as finance or cryptography crates, which makes it harder to find non-cryptocurrency crates in these categories.

This was my reason for adding this category. I'm not a fan of cryptocurrencies at all, but I think it's preferable to have them in one place, where non-interested users won't look, rather than have them pop up all over crates.io.

These crates exist, and need to be somewhere. crates.io can't ban them — many people would find it unacceptable, not because they want cryptocurrency crates, but because they don't want crates.io to be judging usefulness of any technology or gatekeep who can use Rust.

I am concerned that existence of this category can be perceived as an endorsement. But as you're aware, having a category *and* describing it negatively is also controversial.

So I don't see a winning move here. Switch to blessed tags doesn't change anything. It will be as controversial whether "blockchain" or "cryptocurrency" tag deserves to be highlighted. And if they won't be highlighted, then authors will have an incentive to also tag their crates with "cryptography", "finance" or other more visible tags.
